src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -195,131 +201,76 @@ export const MultiSelect = chakra(
         setItemsList(filteredItems);
       };
 
-      // The clearSearchKeyword method resets the search keyword in the MultiSelect component.
-      // If the listOverflow is set to "scroll," it sets the list of items to the full list (items).
-      // If listOverflow is set to a value other than "scroll,"
-      // it checks if the current count of displayed items (listItemsCount) is equal to the default number of visible items (defaultItemsVisible).
-      // If true, it displays the default items; otherwise, it sets the list of items to the full list.
-      // This method is responsible for clearing the search and displaying the appropriate items based on the listOverflow and item count conditions.
       const clearSearchKeyword = () => {
-        if (listOverflow === "scroll") {
-          setItemsList(items);
-        } else {
-          listItemsCount === defaultItemsVisible
-            ? displayDefaultItems()
-            : setItemsList(items);
-        }
+        isExpandable ? setItemsList(defaultItemsList) : setItemsList(items);
       };
 
-      /**
-       * Display default items based on the specified limit.
-       * This function creates a list of items to be displayed, limited by the `defaultItemsVisible` parameter.
-       * The resulting list is set using the `setItemsList` function.
-       */
-      const displayDefaultItems = useCallback(() => {
-        const list = [];
-        let count = 0;
-        for (let i = 0; i < items.length && count < defaultItemsVisible; i++) {
-          const currentItem = items[i];
-          count++;
-          list.push(currentItem);
-          if (count >= defaultItemsVisible) {
-            break;
-          }
-        }
-        setItemsList(list);
-      }, [items, defaultItemsVisible, setItemsList]);
+      /** Toggle for listOverflow = "expand" */
+      const toggleItemsList = () => {
+        setIsExpandable((prevProp) => !prevProp);
+      };
 
-      // Separate effect for handling listOverflow "expand"
       React.useEffect(() => {
-        if (listOverflow === "expand") {
-          setListHeight("");
-          if (listItemsCount === defaultItemsVisible) {
-            displayDefaultItems();
-          } else {
-            setItemsList(items);
-            setViewAllLabel("View less");
-          }
-        }
-      }, [
-        listOverflow,
-        listItemsCount,
-        items,
-        defaultItemsVisible,
-        displayDefaultItems,
-      ]);
+        setItemsList(isExpandable ? defaultItemsList : items);
+      }, [isExpandable]);
```

Comment:
Include the other variables into the dep array. This is the Github linting error:

![Screen Shot 2024-02-16 at 4 36 50 PM](https://github.com/NYPL/nypl-design-system/assets/1280564/eae6fc6d-473b-4e4a-8cdb-2b5977f2ee7e)

But this should also appear locally if eslint is set up:

<img width="926" alt="Screen Shot 2024-02-16 at 4 38 19 PM" src="https://github.com/NYPL/nypl-design-system/assets/1280564/8debe294-d421-43ce-921a-550d691ed231">